config/cmake/addGitHooks.cmake

```
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 2.8.7)
+
+find_package(Git)
+if(NOT GIT_FOUND)
+    message(FATAL_ERROR "git not found!")
+endif()
+
+include(clang-cxx-dev-tools)
+
+function(set_git_hooks_enabled)
+    execute_process(
+            COMMAND ${GIT_EXECUTABLE} config --local core.hooksPath .githooks
+    )
+endfunction()
+
+function(set_git_hooks_disabled)
+    execute_process(
+            COMMAND ${GIT_EXECUTABLE} config --local --unset core.hooksPath
+    )
+endfunction()
+
+option(ENABLE_GIT_HOOKS "Activate git hooks to run clang-format on committed files." ON)
+if (ENABLE_GIT_HOOKS)
+    if (CLANG_FORMAT)
```

Should probably also make sure the HELICS_ENABLE_CLANG_TOOLS option is enabled, since that's the option that includes the clang tools earlier in the root CMakeLists.txt file.
```suggestion
    if (HELICS_ENABLE_CLANG_TOOLS AND CLANG_FORMAT)
```

---

`HELICS_ENABLE_CLANG_TOOLS` is currently always `OFF`, and only works if the compiler is `Clang`, looking at it a bit closer now, depending on this flag is problematic. it may make more sense to simply verify `clang-format` is available separately for this code.

---

The clang tools targets probably need some updating since I haven't used it in a while.  So I don't think a dependency for this work is necessary or even desirable if we are adding a commit hook and auto formatting PR generator in #930